### PR TITLE
The User IP added to the JWT token's payload. In addition, the IP is …

### PR DESCRIPTION
**Pull request by dz-ai**
@ Hitansh159
The User IP added to the JWT token's payload. In addition, the IP is encrypted, so as it go back to user browser it will not be exposes.

I change 3 files:
In Util - added encryption and decryption functions to encrypt/decrypt IP (can be use for other things then IP too).
And added the encryption function to generateToken function to create encrypted IP, and add the encrypted IP to JWT payload.
Fix typo from genrateToken() {} to gen(e)rateToken() {}.

In controllers/auth.js - add req.ip as generateToken argument.

In middleware/auth.js - I change the JWT verify to be inside try catch and added the logic of the IP decryption, and the compare it with the current req.ip of this req in order to approve that the req came from the token owner and from the same machine.

### CODE REVIEW

File: middleware/auth.js [hunk 1]
```diff
@@ -1,17 +1,24 @@
...
-    if (err) return res.sendStatus(403);
-
-    req.user = user;
+  try {
+    const payload = jwt.verify(token, process.env.TOKEN_SECRET);
+    const decryptedIP = decrypt(payload.ip);
+    if (decryptedIP !== req.ip) {
+      res.status(403)
+      res.send({err:"Unauthorized"});
+    }
```
Comment (Hitansh159) [inline middleware/auth.js:16]: add user to details to request when IP matches 
req.user= payload

**Review by Hitansh159 — APPROVED** [review]

File: middleware/auth.js (hunk 1 above)
Comment (dz-ai) [reply middleware/auth.js:16]: Done.

**Comment by dz-ai** [comment]
I see that you make development branch to be base branch.  
shoul I pull now changes from this branch? (in order to Resolve the conflict).

**Comment by dz-ai** [comment]
@TejasNair9977 
@Shivam-53
Conflict Resolve

**Comment by Hitansh159** [comment]
Thanks for Contributing 
we had some blockers past some time. Blockers are removed so if you wish you can continue contributing to future issues.

**Pull request merged**